crates/storage/provider/src/traits/block.rs

```diff
@@ -12,6 +12,16 @@ use reth_primitives::{
 };
 use std::ops::RangeInclusive;
 
+/// Enum to control transaction hash inclusion.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Default)]
+pub enum TransactionKind {
+    /// Indicates that transactions should be processed without including their hashes.
+    #[default]
+    NoHash,
+    /// Indicates that transactions should be processed along with their hashes.
+    WithHash,
```

Comment:
I think the default should be with hash, because not hash should require some thought

Reply:
@mattsse Done